app/models/setting/content.rb

```diff
@@ -147,7 +147,11 @@ def self.default_settings
       self.set('host_tasks_workers_pool_size', N_("Amount of workers in the pool to handle the execution of host-related tasks. When set to 0, the default queue will be used instead. Restart of the dynflowd/foreman-tasks service is required."),
                5, N_('Host Tasks Workers Pool Size')),
       self.set('applicability_batch_size', N_("Number of host applicability calculations to process per task."),
-               50, N_('Applicability Batch Size'))
+               50, N_('Applicability Batch Size')),
+      self.set('autosearch_while_typing', N_('For pages that support it, automatically perform search while typing in search input.'),
+               true, N_('Autosearch')),
+      self.set('autosearch_delay', N_('If Autosearch is enabled, delay in milliseconds before executing searches while typing.'),
+               500, N_('Autosearch delay'))
```

Comment:
TBH i don't think these should be configurable settings. The delay should be set to a reasonable default and the auto search should only be enabled on pages where it can return result in a timely fashion without hammering the server with requests. (i.e. <100 ms to get results for a full db)

Reply:
@tbrisker I am planning on running some production tests to see how well the autosearch performs on a servers with a lot of Content Views. I think that will allow us to make better decisions here on whether A) Autosearch should be enabled by default and B) what the delay should be. It's possible that we don't enable autosearch by default if it is not performant.

I agree we could do without the delay setting, but it would be nice to see how things perform in production first. I'm hesitant to remove it now because of how experimental this feature is. Does that sound ok to you?

Reply:
My opinion is that since these settings only affect an experimental page while cluttering the settings page for all users, my preference would be to drop them completely and try to go with "[hopefully] sane default" first and only add a setting if needed in the future and we can't settle on a default that works for all users.